src/dashboard/src/components/filesystem_ajax/views.py

```diff
@@ -114,13 +114,23 @@ def contents(request):
     return helpers.json_response(response)
 
 def arrange_contents(request):
-    base_path = base64.b64decode(request.GET.get('path', DEFAULT_ARRANGE_PATH))
-
-    # Must indicate that base_path is a folder by ending with /
-    if not base_path.endswith('/'):
-        base_path += '/'
+    path = request.GET.get('path')
+    if path is not None:
+        try:
+            base_path = base64.b64decode(path)
+        except TypeError:
+            response = {
+                'success': False,
+                'message': 'Could not base64-decoded provided path: ' + path,
+            }
+            helpers.json_response(response, status_code=400)
```

Comment:
https://docs.djangoproject.com/en/1.8/ref/request-response/#jsonresponse-objects :o Probably not for this PR but I'd like to replace `helpers.json_response` with ` JsonResponse`